can/interfaces/socketcan/socketcan.py

```diff
@@ -376,7 +380,9 @@ def _tx_setup(
         for message in messages:
             body += build_can_frame(message)
         log.debug("Sending BCM command")
-        send_bcm(self.bcm_socket, header + body)
+
+        if send:
```

Comment:
I think this check should be in some other place, but the whole class needs some refactoring. I'll fix this later.

Reply:
Se the comment above